dspace-api/src/main/java/org/dspace/external/provider/impl/SHERPAv2PublisherDataProvider.java

```diff
@@ -0,0 +1,361 @@
+                } finally {
+                    if (content != null) {
+                        content.close();
+                    }
+                }
+            } else {
+                sherpaResponse = new SHERPAPublisherResponse("SHERPA/RoMEO returned no response");
+            }
+        } catch (Exception e) {
+            log.error("SHERPA/RoMEO query failed: ", e);
+        }
+
+        if (sherpaResponse == null) {
+            sherpaResponse = new SHERPAPublisherResponse("Error processing the SHERPA/RoMEO answer");
+        }
+        if (CollectionUtils.isNotEmpty(sherpaResponse.getPublishers())) {
+            SHERPAPublisher sherpaPublisher = sherpaResponse.getPublishers().get(0);
+            // Construct external data object from returned publisher
+            ExternalDataObject externalDataObject = constructExternalDataObjectFromSherpaPublisher(sherpaPublisher);
+            return Optional.of(externalDataObject);
+        }
+        return Optional.empty();
+    }
+
+    /**
+     * Search SHERPA v2 API for publisher results based on a 'contains word' query for publisher name
+     * @param query The query for the search
+     * @param start The start of the search
+     * @param limit The max amount of records to be returned by the search
+     * @return a list of external data objects
+     */
+    @Override
+    public List<ExternalDataObject> searchExternalDataObjects(String query, int start, int limit) {
+        // query args to add to SHERPA/RoMEO request URL
+        HttpGet get = null;
+        try {
+            get = constructHttpGet(query, start, limit);
+            HttpClient hc = new DefaultHttpClient();
+            HttpResponse response = hc.execute(get);
+            if (response.getStatusLine().getStatusCode() == 200) {
+                // Get response input stream and parse JSON into new SHERPAResponse object
+                InputStream content = null;
+                try {
+                    content = response.getEntity().getContent();
+                    SHERPAPublisherResponse sherpaResponse =
+                        new SHERPAPublisherResponse(content, SHERPAPublisherResponse.SHERPAFormat.JSON);
+                    if (CollectionUtils.isNotEmpty(sherpaResponse.getPublishers())) {
+                        List<ExternalDataObject> list = sherpaResponse.getPublishers().stream().map(
+                            sherpaPublisher -> constructExternalDataObjectFromSherpaPublisher(sherpaPublisher)).collect(
+                            Collectors.toList());
+
+                        // Unlike the previous API version we can request offset and limit, so no need to build a
+                        // sublist from this list, we can just return the list.
+                        return list;
+                    }
+                } catch (IOException e) {
+                    log.error("Error parsing SHERPA response input stream: " + e.getMessage());
```

Comment:
I have now rethrown this exception in 3ef47dac - this one I can do, as it is nested within a higher try/catch block.
However, note that for the main catch in the method:
This class and its methods implement ExternalDataService interface (via extending ExternalDataServiceImpl). The abstract methods don't have any throw statements in its signatures. I would be hesitant to include a refactor of this interface and abstract impl in this PR as it would have consequences for all other providers.